File: JAVA/Day2/TestTask4.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;

public class TestTask4 {

    @Test
    public void testEnoughFuelExactDistance() {
        boolean result = Task4.isPossibleOrNot(50, 25, 2);
        assertTrue(result);
    }

    @Test
    public void testNotEnoughFuel() {
        boolean result = Task4.isPossibleOrNot(100, 25, 2);
        assertFalse(result);
    }

    @Test
    public void testMoreThanEnoughFuel() {
        boolean result = Task4.isPossibleOrNot(40, 25, 2);
        assertTrue(result);
    }

    @Test
    public void testZeroFuel() {
        boolean result = Task4.isPossibleOrNot(10, 25, 0);
        assertFalse(result);
    }
}

File: Python/Day2/TestTask4.py
import unittest
import Task4

class TestTask4(unittest.TestCase):
    def test_enough_fuel_exact_distance(self):
        result = Task4.is_possible_or_not(50, 25, 2)
        self.assertTrue(result)

    def test_not_enough_fuel(self):
        result = Task4.is_possible_or_not(100, 25, 2)
        self.assertFalse(result)

    def test_more_than_enough_fuel(self):
        result = Task4.is_possible_or_not(40, 25, 2)
        self.assertTrue(result)

    def test_zero_fuel(self):
        result = Task4.is_possible_or_not(10, 25, 0)
        self.assertFalse(result)